### ISSUE TITLE
Traits not checking for Self: Sized on required methods

### ISSUE THREAD

**Issue by ionicmc-rs** [rust-lang/rust#134422]
I tried this code:

```rust
trait MyTrait {
    fn foo(self);
}
```

I expected to see this happen:
```shell
Error: the size for values of type 'Self' do not have a know size at compile time
# ...
```

Instead, this happened:
```
# No Errors
```

### Meta

`rustc --version --verbose`:
```
rustc 1.85.0-nightly (21fe748be 2024-12-11)
binary: rustc
commit-hash: 21fe748be15271ea5804e0507cd699b675efe038
commit-date: 2024-12-11
host: x86_64-unknown-linux-gnu
release: 1.85.0-nightly
LLVM version: 19.1.5
```

<details><summary>Backtrace</summary>
<p>

```
# None, because it does not throw an error, which is kinda the problem
```

</p>
</details>

The only reason i think this is here is to allow for `dyn` compatibility

**Comment by compiler-errors** [comment]
This is not a diagnostics bug. This is also not something that can be changed -- if we began to enforce this, we'd be breaking existing code.

**Comment by compiler-errors** [comment]
Actually, I'm just gonna close this b/c it's really not actionable. Thanks for the issue report anyways.
